File: abc_training_addition/C_diff/C_Chokudai.py
# C - Chokudai
# ----------------------------------------
# 問題
# https://atcoder.jp/contests/abc211/tasks/abc211_c

# 参考
# https://atcoder.jp/contests/abc211/editorial/2287

# かなり難しかった、DPに慣れる必要がある

# AC (解説)
# ----------------------------------------

### LCS によるアプローチ
# def exprint(x): print(*x, sep="\n")
# def init_array(i, j, val=0): return [[val]*j for _ in range(i)]

# ceo = "chokudai"
# S = input()

# DP = init_array(len(ceo)+1, len(S)+1)

# counter = 0
# for i in range(len(ceo)):
#     for j in range(len(S)):
#         if ceo[i] == S[j]:
#             counter += 1
#             DP[i+1][j+1] = DP[i][j] + 1
#         else:
#             DP[i+1][j+1] = max(
#                 DP[i][j+1],
#                 DP[i+1][j]
#             )

# exprint(DP)
# print(counter)
# -> だめですね

### 解説
def exprint(x): print(*x, sep="\n")
def init_array(i, j, val=0): return [[val]*j for _ in range(i)]

ceo = "chokudai"
S = input()
lc, ls = len(ceo), len(S)

MOD = int(1e9 + 7)
DP = [[1] + [0]*lc for _ in range(ls+1)]

for i in range(1, ls+1):
    for j in range(1, lc+1):

        if S[i-1] == ceo[j-1]:  # ...(i)
            DP[i][j] = ( DP[i-1][j] + DP[i-1][j-1] ) % MOD 
        else:                   # ...(ii)
            DP[i][j] = DP[i-1][j]


print(DP[-1][-1])

### 解釈
# (i)について
#    S[i]に下線を引くことはありえないためDP[i-1][j]の結果を引き継ぐ
# (ii)について
#    - S[i]に下線を引かない場合
#      (i)と同様にDP[i-1][j]に等しい
#    - S[i]に下線を引く場合
#      Sの(i-1)文字目までを使ってTの(i-1)文字目までを選択する方法　の場合の数に等しい
#      よってDP[i-1][j-1]
#    上2つの和である.

File: contests/abc211/C_chokudai.rs
//               C - chokudai
// ----------------------------------------
// 問題
// https://atcoder.jp/contests/abc211/tasks/abc211_c
// ----------------------------------------

// attributes
#![allow(unused_imports)]
#![allow(unused_variables)]
#![allow(non_snake_case)]
#![allow(dead_code)]
#![allow(unused_macros)]

// imports
use proconio::{
    fastout, input,
    marker::{Bytes, Chars, Usize1},
};
use std::cmp::{max, min, Reverse};
use std::collections::{BTreeMap, BTreeSet, BinaryHeap, HashMap, HashSet, VecDeque};

// constant
const MOD: usize = 1_000_000_007;
const INF: usize = 1001001001001001001;
const NEG1: usize = 1_usize.wrapping_neg();
const CHOKUDAI: &str = "chokudai";

macro_rules! set_madd {
    ( $a:expr, $($rem:expr),+ $(,)* ) => {{
        let sum = madd!( $($rem),+ );
        $a = ($a + sum) % MOD;
    }};
}

macro_rules! madd {
    ( $a:expr ) => {{
        $a % MOD
    }};
    ( $a:expr, $b:expr ) => {{
        ($a + $b) % MOD
    }};
    ( $a:expr, $($rem:expr),+ $(,)* ) => {{
        ($a + madd!( $($rem),+) ) % MOD
    }};
}

// solve
fn main() {
    input! {
        S: Chars,
    }
    let C: Vec<char> = CHOKUDAI.chars().collect();

    let mut dp = vec![vec![0; 9]; S.len() + 1];
    for i in 0..=S.len() {
        dp[i][0] = 1;
    }

    // dp更新
    for i in 1..=S.len() {
        for j in 1..=8 {
            if S[i-1] == C[j-1] {
                set_madd!(
                    dp[i][j],
                    dp[i-1][j],
                    dp[i-1][j-1],
                );
            } else {
                dp[i][j] = dp[i-1][j];
            }
        }
    }

    println!("{}", dp[S.len()][8]);
}
